### PR TITLE
Clarify where secret expiration date is and add troubleshooting section

### PR DESCRIPTION
**Pull request by soltysh**
Follow up to #3436. I've clarified where to find the secret expiry annotation and I've added troubleshooting section how to deal with possible problems.

@ahardin-rh ptal
@geliu2016 fyi

### CODE REVIEW

File: dev_guide/secrets.adoc [hunk 1]
```diff
@@ -242,3 +243,35 @@ spec:
...
+|Issue |Resolution
+a|A xref:service-serving-certificate-secrets[service certificate generations]
+fails with (service's `*service.alpha.openshift.io/serving-cert-generation-error*`
+annotation contains):
+
+----
+secret/ssl-key references serviceUID 62ad25ca-d703-11e6-9d6f-0e9c0057b608, which does not match 77b6dd80-d716-11e6-9d6f-0e9c0057b60
+----
+a|The service that generated the ceritiface no longer exists (has different
+`*serviceUID*`).  You need to force certificates regeneration by removing the
```
Comment (ahardin-rh) [inline dev_guide/secrets.adoc]: s/need to/must

**Comment by ahardin-rh** [comment]
Just one nit from me. Thanks!

**Comment by soltysh** [comment]
Updated.

**Comment by ahardin-rh** [comment]
No revision history needed

**Pull request merged**